src/main/scala/it/pagopa/interop/agreementprocess/service/AgreementContractCreator.scala

```diff
@@ -119,14 +124,26 @@ final class AgreementContractCreator(
 
   }
 
-  def getSubmissionInfo(
-    seed: UpdateAgreementSeed
-  )(implicit contexts: Seq[(String, String)], ec: ExecutionContext): Future[(String, OffsetDateTime)] =
+  def getSubmissionInfo(seed: UpdateAgreementSeed, consumer: PersistentTenant, producer: PersistentTenant)(implicit
+    contexts: Seq[(String, String)],
+    ec: ExecutionContext
+  ): Future[(String, OffsetDateTime)] =
     for {
-      selfcareUuidd   <- getSelfcareIdFutureUUID(contexts)
-      submission      <- seed.stamps.submission.toFuture(StampNotFound("submission"))
-      userResponse    <- selfcareV2ClientService.getUserById(selfcareUuidd, submission.who).map(_.toApi)
-      userResponseApi <- userResponse.toFuture.recoverWith { case _ => Future.failed(MissingUserInfo(submission.who)) }
+      submission                 <- seed.stamps.submission.toFuture(StampNotFound("submission"))
+      consumerSelfcareId         <- consumer.selfcareId.toFuture(SelfcareIdNotFound(consumer.id))
+      producerSelfcareId         <- producer.selfcareId.toFuture(SelfcareIdNotFound(producer.id))
+      consumerSelfcareUuid       <- consumerSelfcareId.toFutureUUID
+      producerSelfcareUuid       <- producerSelfcareId.toFutureUUID
+      consumerSelfcare           <- selfcareV2ClientService.getInstitution(consumerSelfcareUuid).map(_.toApi)
+      producerSelfcare           <- selfcareV2ClientService.getInstitution(producerSelfcareUuid).map(_.toApi)
+      (consumerApi, producerApi) <- consumerSelfcare.toFuture.zip(producerSelfcare.toFuture)
+      userResponse               <- selfcareV2ClientService
+        .getUserById(consumerSelfcareUuid, consumerApi.id)
```

Comment:
I updated the ticket, the submission can be performed only by the consumer, it is enough to retrieve from selfcare the user info always in relation to the consumer